pkg/asset/installconfig/installconfig.go

```diff
@@ -156,3 +156,16 @@ func parseCIDR(s string) net.IPNet {
 	_, cidr, _ := net.ParseCIDR(s)
 	return *cidr
 }
+
+// Load returns the installconfig from disk.
+func (a *InstallConfig) Load(p asset.FileFetcher) (found bool, err error) {
```

Comment:
It is not sufficient to fill out `a.file`. On load, we need to fill out `a.config` as well. Many of the assets that depend upon this asset will get data from `a.config`.

Reply:
This is not good, looks like the tfvar has a field `platfom`, which cannot be derived from the on-disk files directly.

Reply:
I wonder if we are doing that in the TerraformVariables asset solely so that the Cluster asset does not have to depend upon the InstallConfig asset? And whether that is so that me can remove the diamonds from the dependency graph? Since we have a lot of diamonds currently, I suggest that for the sake of landing this PR we have Cluster get the platform from InstallConfig. We can remove diamonds as a later PR: And if we do, then we should remove all diamonds.

Reply:
Although we should make that decision soon. Because if we *are* going to remove diamonds, then we don't need to store the state file.